instrumentation/resources/library/src/test/java/io/opentelemetry/instrumentation/resources/ContainerResourceTest.java

```diff
@@ -7,92 +7,82 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
 
 class ContainerResourceTest {
 
-  @Test
-  void buildResource_Invalid(@TempDir Path tempFolder) throws IOException {
-    // invalid containerId (non-hex)
-    Path cgroup =
-        createCgroup(
-            tempFolder.resolve("cgroup1"),
-            "13:name=systemd:/podruntime/docker/kubepods/ac679f8a8319c8cf7d38e1adf263bc08d23zzzz");
-    assertThat(buildResource(cgroup)).isEqualTo(Resource.empty());
-
-    // unrecognized format (last "-" is after last ".")
-    cgroup =
-        createCgroup(
-            tempFolder.resolve("cgroup1"),
-            "13:name=systemd:/podruntime/docker/kubepods/ac679f8.a8319c8cf7d38e1adf263bc08-d23zzzz");
-    assertThat(buildResource(cgroup)).isEqualTo(Resource.empty());
-
-    // test invalid file
-    cgroup = tempFolder.resolve("DoesNotExist");
+  @ParameterizedTest
+  @ValueSource(
+      strings = {
+        // invalid containerId (non-hex)
+        "13:name=systemd:/podruntime/docker/kubepods/ac679f8a8319c8cf7d38e1adf263bc08d23zzzz",
+        // unrecognized format (last "-" is after last ".")
+        "13:name=systemd:/podruntime/docker/kubepods/ac679f8.a8319c8cf7d38e1adf263bc08-d23zzzz"
+      })
+  void buildResource_returnsEmptyResource_whenContainerIdIsInvalid(
+      String line, @TempDir Path tempFolder) throws IOException {
+    Path cgroup = createCgroup(tempFolder.resolve("cgroup"), line);
     assertThat(buildResource(cgroup)).isEqualTo(Resource.empty());
   }
 
   @Test
-  void buildResource_Valid(@TempDir Path tempFolder) throws IOException {
-    // with suffix
-    Path cgroup =
-        createCgroup(
-            tempFolder.resolve("cgroup1"),
-            "13:name=systemd:/podruntime/docker/kubepods/ac679f8a8319c8cf7d38e1adf263bc08d23.aaaa");
-    assertThat(getContainerId(buildResource(cgroup)))
-        .isEqualTo("ac679f8a8319c8cf7d38e1adf263bc08d23");
-
-    // with prefix and suffix
-    Path cgroup2 =
-        createCgroup(
-            tempFolder.resolve("cgroup2"),
-            "13:name=systemd:/podruntime/docker/kubepods/crio-dc679f8a8319c8cf7d38e1adf263bc08d23.stuff");
-    assertThat(getContainerId(buildResource(cgroup2)))
-        .isEqualTo("dc679f8a8319c8cf7d38e1adf263bc08d23");
+  void buildResource_returnsEmptyResource_whenFileDoesNotExist(@TempDir Path tempFolder) {
+    Path cgroup = tempFolder.resolve("DoesNotExist");
+    assertThat(buildResource(cgroup)).isEqualTo(Resource.empty());
+  }
 
-    // just container id
-    Path cgroup3 =
-        createCgroup(
-            tempFolder.resolve("cgroup3"),
-            "13:name=systemd:/pod/d86d75589bf6cc254f3e2cc29debdf85dde404998aa128997a819ff991827356");
-    assertThat(getContainerId(buildResource(cgroup3)))
-        .isEqualTo("d86d75589bf6cc254f3e2cc29debdf85dde404998aa128997a819ff991827356");
+  @ParameterizedTest
+  @MethodSource
```

Comment:
Can you use a named `@MethodSource` or a `@ArgumentsSource` instead? I believe it's a bit easier to read and debug this way, as you can easily click on that, and you're not forced to rely on some naming convention.

Reply:
Sure! :)